test/unit/logger/logfile_match_wrapper.cmake

```diff
@@ -0,0 +1,50 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: MIT
+
+#
+# no_logfile.cmake -- wrapper script for tests verifying that logfile was not created
+# pass NO_LOGFILE argument for enabling check that the file should not exist after the test
+#                 is finished
+# pass LOGFILE_MATCH argument for running logfile contents validation
+#
+
+if(NOT DEFINED BIN_PATH)
+    message(FATAL_ERROR "BIN_PATH needs to be defined with a test binary to be ran")
+endif()
+if(NOT DEFINED OUT_FILE)
+    message(FATAL_ERROR "OUT_FILE needs to be defined with an output file to be verified")
+endif()
+if(DEFINED LOGFILE_MATCH AND NOT DEFINED MATCH_FILE)
+    message(FATAL_ERROR "MATCH_FILE needs to be defined with LOGFILE_MATCH enbled")
+endif()
+
+if(EXISTS ${OUT_FILE})
+    file(REMOVE ${OUT_FILE})
+endif()
+
+execute_process(
```

Comment:
I mean, this is exactly what the match.cmake does. So what you are effectively doing is creating a wrapper with the same functionality that is being disabled (by mode 'binary') in the main script?

Again, I'm not sure I fully understand, but if all you are trying to do is match against different output destination (stdout, stderr, file), then it'd be easier to create that argument in the original match script.